File: ps/pub/publisher.go
package main

import (
	"github.com/gofiber/fiber/v2"
	zmq "github.com/pebbe/zmq4"
)

func main() {
	app := fiber.New()
	zctx, _ := zmq.NewContext()

	p, _ := zctx.NewSocket(zmq.PUB)
	p.Bind("tcp://*:5555")

	app.Post("/", func(c *fiber.Ctx) error {
		p.Send("pubsub", zmq.SNDMORE)
		p.Send(string(c.Body()), 0)
		return c.SendString("Sent to the subscriber/worker.")
	})

	app.Listen(":3000")

}

File: ps/pub/publisher.js
const Fast = require("fastify");
const zmq = require("zeromq");

const app = new Fast();
const p = new zmq.Publisher();

app.post("/", async (request, reply) => {
  const msg = JSON.stringify({ ...request.body });
  console.log("Received: %s", msg);

  await p.send(["pubsub", msg]);
  return reply.send("Sent to the subscriber/worker.");
});

const main = async () => {
  try {
    await p.bind("tcp://*:5555");
    await app.listen({ port: 3000 });
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
};
main();
